pktoken/pktoken.go

```diff
@@ -192,31 +198,39 @@ func (p *PKToken) UnmarshalJSON(data []byte) error {
 	cicCount := 0
 	cosCount := 0
 	for _, signature := range parsed.Signatures() {
-		sigHeader, ok := signature.PublicHeaders().Get(sigTypeHeader)
-		if !ok {
-			return fmt.Errorf(`pk token signature is missing required "%s" header as public header`, sigTypeHeader)
+		// for some reason the unmarshaled signatures have empty non-nil
+		// public headers. set them to nil instead.
+		public := signature.PublicHeaders()
+		pubMap, _ := public.AsMap(context.Background())
+		if len(pubMap) == 0 {
+			signature.SetPublicHeaders(nil)
 		}
 
-		sigHeaderString, ok := sigHeader.(string)
+		protected := signature.ProtectedHeaders()
+		typeHeader, ok := protected.Get(jws.TypeKey)
+		if !ok {
+			// TODO: default to OIDC?
+			return fmt.Errorf("missing typ header")
```

Comment:
I believe should default to OIDC here since it is completely within the JWS standard to not include a `typ` header. OpenPubkey should always include a `typ` header for a COS and CIC signature.

It might make sense to log a warning, because it is very uncommon for a OP to not include a `typ` header. It is happens, it is probably a since of some weirdnesss.